### DISCUSSION TITLE
How to get started with building an SCM extension

### DISCUSSION THREAD

**Discussion by GallifreyMage** [microsoft/vscode-discussions#1717] [Extension Development QnA] [answered]
Hi there, I'm still somewhat new to Code, but I've built some syntax extensions using Yo Code.  I work somewhere that has an old but private SCM tool.  I'd like to be able to hook it up to Code, but I don't really know where to get started. I've tried to look at the GitHub for the Git SCM, but I'm just overwhelmed.

Our local tool uses a command line and discrete version numbering.  We have projects and sub-projects defined by a flat-file structure (we call it a scope...).    I can get lists of all the items available and what version they are, and whether or not they live in the repo.

I'm just trying to figure out how to marry the two pieces together.

Oh yeah - and I'm and engineer, not a coder.  I can do python pretty well, and Java Script (tho it's been a minute), but I'm also still messing with Fortran 77...  so my skills are somewhat limited.  So odds are you can't dumb down the answer too much for me.

Many thanks

**Comment by olguzzar** [comment, marked as answer]
Hi there, I recommend checking out the Source Control API [extension guide](https://code.visualstudio.com/api/extension-guides/scm-provider) and [this extension sample](https://github.com/microsoft/vscode-extension-samples/tree/main/source-control-sample) to get started! If you have more specific questions after looking through that, we can help further.

**Comment by robconery** [comment]
Closing as answered - please feel free to reopen if you need more information.

**Reply by GallifreyMage** [reply]
hi @robconery 
I would like to reopen the discussion please, but I do not see a magic button to let me do that.  Is that something I need to ask you to do?

Thanks,
GM

**Comment by GallifreyMage** [comment]
Hi there,
Thank you for responding to my question.
I must admit I'm still a bit stumped. 
I'll start at the beginning. 
I'm struggling with following how to hook up my repo information to the API.

For example, I might call a command line like this:

repo -m modelinfo.mdl

and it would return a list of everything in the repo that that mdl file could find (which would be a subset of the complete repo).

It can be tailored in terms of the information it receives, but generally it would look something like:

```
File Rich.py { loc = aaa.bbb.ccc;  ver=2; }
File Mike.ts { loc = ddd.eee;      ver=3; }
```

or it could be converted to full paths for each file.

I do not understand how to:
 - tell the API what commands to execute to retrieve the information
 - tell the API how to process the information 
 - display that information in the SCM view

At least that's where I'm starting from.  It may fall into place at some point, but I feel like missing some key pieces which I don't see from the examples.

Many thanks,
GallifreyMage
